src/wp-includes/option.php

```diff
@@ -196,6 +169,33 @@ function get_option( $option, $default_value = false ) {
 			$value = wp_cache_get( $option, 'options' );
 
 			if ( false === $value ) {
+				// Prevent non-existent options from triggering multiple queries.
+				$notoptions = wp_cache_get( 'notoptions', 'options' );
+
+				// Prevent non-existent `notoptions` key from triggering multiple key lookups.
+				if ( ! is_array( $notoptions ) ) {
+					$notoptions = array();
```

Comment:
Minor note: Lines 206-208 below also check `! is_array( $notoptions )` and then sets it to an array. The later check seems unnecessary (and appears to have been unnecessary since before this PR actually. Should we clean this up while we're here and dealing with `$notoptions`, or consider it out of scope?

Reply:
Agreed, I think we can safely remove lines 206-208.